src/Microsoft.Health.Fhir.Core.UnitTests/Persistence/TransactionFailedExceptionTests.cs

```diff
@@ -0,0 +1,70 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using Microsoft.Health.Fhir.Core.Features.Persistence;
+using Microsoft.Health.Fhir.Core.Models;
+using Xunit;
+
+namespace Microsoft.Health.Fhir.Core.UnitTests.Persistence
+{
+    public class TransactionFailedExceptionTests
+    {
+        [Fact]
+        public void GivenAListOfOperationOutComeIssues_WhenInitialized_ThenCorrectOperationOutcomeIssuesShouldBeAdded()
+        {
+            string message = "message";
+            HttpStatusCode statusCode = HttpStatusCode.Processing;
+            var operationOutComeIssues = GetOperationOutcomeIssues(message);
+
+            var exception = new TransactionFailedException(message, statusCode, operationOutComeIssues);
+
+            Assert.NotNull(exception.Issues);
+            Assert.Equal(3, exception.Issues.Count);
+
+            var firtIssue = exception.Issues.First();
+            Assert.Equal(OperationOutcomeConstants.IssueSeverity.Error, firtIssue.Severity);
+            Assert.Equal(OperationOutcomeConstants.IssueType.Processing, firtIssue.Code);
+            Assert.Equal(message, firtIssue.Diagnostics);
```

Comment:
nit: could probably have a private method that does the asserts since we're doing the same asserts 4 times in the code.